src/Host/API/Impl/Definitions/IRHostSession.cs

```diff
@@ -0,0 +1,179 @@
+        /// <exception cref="OperationCanceledException" />
+        /// <exception cref="RHostDisconnectedException" />
+        Task<T> EvaluateAsync<T>(string expression, CancellationToken cancellationToken = default(CancellationToken));
+
+        /// <summary>
+        /// Invokes R function. 
+        /// </summary>
+        /// <param name="function">Function name</param>
+        /// <param name="cancellationToken">Cancellation token</param>
+        /// <param name="args">Function arguments</param>
+        /// <exception cref="ArgumentException" />
+        /// <exception cref="REvaluationException" />
+        /// <exception cref="OperationCanceledException" />
+        /// <exception cref="RHostDisconnectedException" />
+        Task InvokeAsync(string function, CancellationToken cancellationToken = default(CancellationToken), params object[] args);
+
+        /// <summary>
+        /// Invokes R function. 
+        /// </summary>
+        /// <param name="function">Function name</param>
+        /// <param name="cancellationToken">Cancellation token</param>
+        /// <param name="args">Function arguments</param>
+        /// <returns>Name of the variable that holds the data returned by the function</returns>
+        /// <exception cref="ArgumentException" />
+        /// <exception cref="REvaluationException" />
+        /// <exception cref="OperationCanceledException" />
+        /// <exception cref="RHostDisconnectedException" />
+        Task<string> InvokeAndReturnAsync(string function, CancellationToken cancellationToken = default(CancellationToken), params object[] args);
+
+        /// <summary>
+        /// Retrieves list of unknown type from R variable
+        /// </summary>
+        /// <param name="expression">Expression (variable name) to fetch as list</param>
+        /// <param name="cancellationToken">Cancellation token</param>
+        /// <returns>List of objects</returns>
+        /// <exception cref="ArgumentException" />
+        /// <exception cref="REvaluationException" />
+        /// <exception cref="OperationCanceledException" />
+        /// <exception cref="RHostDisconnectedException" />
+        Task<List<object>> GetListAsync(string expression, CancellationToken cancellationToken = default(CancellationToken));
+
+        /// <summary>
+        /// Retrieves list of specific type from R variable
+        /// </summary>
+        /// <param name="expression">Expression (variable name) to fetch as list</param>
+        /// <param name="cancellationToken">Cancellation token</param>
+        /// <returns>List of values of the provided type</returns>
+        /// <exception cref="ArgumentException" />
+        /// <exception cref="REvaluationException" />
+        /// <exception cref="OperationCanceledException" />
+        /// <exception cref="RHostDisconnectedException" />
+        Task<List<T>> GetListAsync<T>(string expression, CancellationToken cancellationToken = default(CancellationToken));
```

Comment:
It explicitly forces `as.list` and converts returned `JArray` to the specified type.